### DISCUSSION TITLE
Features that edit rc files (& how to work with dotfiles)

### DISCUSSION THREAD

**Discussion by joshuanianji** [devcontainers/spec#371] [Q&A] [answered]
Some features edit `.bashrc` and `.zshrc`, such as [stuartleeks/shell-history](https://github.com/stuartleeks/dev-container-features/blob/main/src/shell-history/README.md) which needs to `chown` a folder that is mounted.

The issue I'm experiencing comes when I use my custom dotfiles repo, which replaces any `.zshrc` files with a custom one, so any changes made from features are basically deleted.

Do you have any idea on how to fix this? I could source the old `.zshrc` in my custom file, but I'm not sure if that's a good idea. I was also thinking that there can be a "best practice" for devcontainer features to add stuff to `.zshrc.devcontainers` (and so on for bash and fish) so people can take this into account when making install scripts for devcontainer dotfiles.

**Comment by eljog** [comment, marked as answer]
Thank you for starting this discussion.

I see that [stuartleeks/shell-history](https://github.com/stuartleeks/dev-container-features/blob/main/src/shell-history/README.md) is a community contributed feature. Could you please open an issue on their repository [here](https://github.com/stuartleeks/dev-container-features/issues) to let the author/community know?

**Reply by haticecan37** [reply]
> Thank you for starting this discussion.
> 
> 
> 
> I see that [stuartleeks/shell-history](https://github.com/stuartleeks/dev-container-features/blob/main/src/shell-history/README.md) is a community contributed feature. Could you please open an issue on their repository [here](https://github.com/stuartleeks/dev-container-features/issues) to let the author/community know?

**Reply by joshuanianji** [reply]
Thank you! After some more research, I think features that mount volumes should `chown` during the `postStartCommand` or another lifecycle script, instead of adding stuff to bashrc in the install script.

Since this is dependent on the individual feature and not spec, I'll probably add an issue there.
